arithmetization/src/test/java/net/consensys/linea/zktracer/precompiles/LowGasStipendPrecompileCallTests.java

```diff
@@ -0,0 +1,327 @@
+   * @param gasCase the gas case (zero, one, cost minus one, cost, cost plus one).
+   * @param modexpCostGT200 flag indicating if the MODEXP cost is greater than 200. It is ignored
+   *     for other precompile contracts.
+   */
+  @ParameterizedTest
+  @MethodSource("lowGasStipendPrecompileCallTestSource")
+  void lowGasStipendPrecompileCallTest(
+      Address precompileAddress,
+      ArgumentCase argumentCase,
+      GasCase gasCase,
+      boolean modexpCostGT200) {
+    final BytecodeCompiler program = BytecodeCompiler.newProgram();
+
+    // In order to actually trigger the insufficient we need to:
+    // - Set a specific callDataSize for BLAKE2F and EC_PAIRING
+    // - Set the r value of BLAKE2F to have precompileCost > gasBonus
+    // - Populate the memory with a large enough number of words for SHA256, RIPEMD160, and ID
+    //   to have precompileCost > gasBonus.
+    final int value = argumentCase.isZeroCase() ? 0 : 1;
+    final int callDataSize; // depends on the called precompile
+    final int callDataOffset = 0;
+
+    // retSize is defined below
+    final int returnAtOffset = 13;
+
+    // BLAKE2F specific parameters
+    final int rLeadingByte = argumentCase.isZeroCase() ? 0 : 0x12;
+    final int r = rLeadingByte << 8;
+
+    // MODEXP specific parameters
+    final int bbs = modexpCostGT200 ? 0x01 : 0x02;
+    final int ebs = modexpCostGT200 ? 0x06 : 0x03;
+    final int mbs = modexpCostGT200 ? 0x19 : 0x04;
+    BigInteger exponentLog = BigInteger.ZERO;
+
+    // Prepare the arguments for the different precompile calls
+    if (precompileAddress == BLAKE2B_F_COMPRESSION) {
+      callDataSize = PRC_BLAKE2F_SIZE;
+      prepareBlake2F(program, rLeadingByte, callDataOffset);
+    } else if (precompileAddress == ALTBN128_PAIRING) {
+      callDataSize = PRC_ECPAIRING_SIZE;
+    } else if ((precompileAddress == SHA256
+            || precompileAddress == RIPEMD160
+            || precompileAddress == ID)
+        && argumentCase.isNonZeroCase()) {
+      final int nWords = 1024;
+      callDataSize = nWords * WORD_SIZE; // This guarantees that precompileCost > gasBonus
+      prepareSha256Ripemd160Id(program, nWords, callDataOffset);
+    } else if (precompileAddress == MODEXP) {
+      callDataSize = 96 + bbs + ebs + mbs;
+      exponentLog = prepareModexp(bbs, mbs, ebs, callDataSize, program);
+    } else {
+      // ECADD, ECMUL, ECRECOVER cases
+      callDataSize = argumentCase.isZeroCase() ? 0 : 1;
+    }
+
+    // Compute the return size
+    final int returnAtCapacity = getReturnAtCapacity(precompileAddress, callDataSize, mbs);
+
+    // Compute the precompile cost
+    final int precompileCost =
+        getPrecompileCost(precompileAddress, callDataSize, bbs, mbs, exponentLog, r);
+
+    // Compute the gas stipend in the different testing scenarios
+    int gas = getGas(gasCase, precompileCost);
+
+    // In case funds are sent to the precompile contract (argumentCase == NON_ZERO)
```

Comment:
So `ArgumentCase` refers to `value` I take it ? In that case why not call it `ValueParameter` or something similar ?